packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -83,6 +84,16 @@ export class ProfilesCache {
         return require("@zowe/secrets-for-zowe-sdk").keyring;
     }
 
+    public setConfigArray(extendermetadata: zowe.imperative.ICommandProfileTypeConfiguration[]): void {
+        extendermetadata?.forEach((item) => {
+            this.profileTypeConfigurations.push(item);
+        });
+    }
```

Comment:
If the method is named `setConfigArray`, I think it should replace the contents of the entire array. Since we are appending items to the array instead, perhaps it could be called `updateConfigArray` or `addToConfigArray` instead?

Also are we certain that this method won't be called more than once for the same profile type? Wondering if we should check to see whether an item already exists in the array before pushing a duplicate item (or we could use a `Set` for this).

Reply:
Handled duplicate item. Made the required changes.